paasta_tools/contrib/service_shard_update.py

```diff
@@ -194,14 +191,28 @@ def main(args):
 
                     instance_config = {
                         "deploy_group": f"{deploy_prefix}.{args.shard_name}",
-                        "min_instances": args.min_instance_count,
-                        "max_instances": args.prod_max_instance_count
-                        if deploy_prefix == "prod"
-                        else args.non_prod_max_instance_count,
```

Comment:
ah cool, so one thing we should make sure to look at is what paasta will do if there's no min/max instances set, but autoscaling is configured - iirc, it might take until the first autotune run for min/max instances to be set (and therefore for the instance to actually be autoscaled)

if so, i think we need to update what paasta defaults to if there an autoscaling section set and no min/max (explicitly or in autotune)

Reply:
@karmeleon looks like paasta will not setup an HPA (the thing that actually controls autoscaling) and will default to running a single instance

i *think* that once autotune runs, it will set a value for min/max_instance tho

that said: i think we can tweak paasta's behavior here